src/command_modules/azure-cli-interactive/azclishell/frequency_heuristic.py

```diff
@@ -0,0 +1,63 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import datetime
+import json
+
+from azure.cli.core._config import set_global_config_value
+
+from azclishell.configuration import CONFIGURATION, get_config_dir as shell_config
+
+SHELL_CONFIG = CONFIGURATION
+DAYS_AGO = 28
+ACTIVE_STATUS = 5
+DISPLAY_TIME = 20
+
+
+def day_format(now):
+    """ returns the date format """
+    return now.strftime("%Y-%m-%d")
+
+
+def update_frequency():
+    """ updates the frequency from files """
+    if os.path.exists(os.path.join(shell_config(), SHELL_CONFIG.get_frequency())):
```

Comment:
Save `os.path.join(shell_config(), SHELL_CONFIG.get_frequency())` in a variable. It's used several times in this method.